server/lib/pencil_space_server/game.ex

```diff
@@ -0,0 +1,58 @@
+defmodule PencilSpaceServer.Game do
+  @moduledoc """
+  Coordinate the game overall: state, updates, process life-cycles etc.
+  """
+
+  alias PencilSpaceServer.Game.{Monitor, Supervisor}
+
+  def start do
+    name = Nanoid.generate()
+    case Supervisor.supervise({Monitor, name: ref(name)}) do
+      {:ok, _pid} -> {:ok, name}
+      {:error, _} -> {:error}
+    end
+  end
+
+  def presence(name) do
+    case whereis(name) do
+      nil -> {:error, :does_not_exist}
+      __g -> {:ok, :exists}
+    end
+  end
+
+  @doc """
+  Synchronously updates the host on the Game state.
+
+  Returns a Game.State.
+
+  ## Examples
+
+      iex> PencilSpaceServer.Game.update("name-of-game", %{host: 1})
+      %PencilSpaceServer.Game.State{host: 1}
+  """
+  def update(name, %{"host" => host}) do
+    GenServer.call(ref(name), {:host, host})
```

Comment:
This maybe a leaky abstraction. So now the `Game` module knows there is a GenServer running somewhere which handles this message. A more ideal approach would be to change this to `Monitor.update(ref(name), {:host, host})`

That way later on if you wish to move away from a GenServer to some other behaviour, the change would just be in the `Monitor` module.

So I would rather do it by moving this line to `Monitor` module and expose it as a function, which can then be called from `Game` module